xtext-website/documentation/303_runtime_concepts.md

```diff
@@ -249,6 +252,26 @@ override getScope(EObject context, EReference reference) {
 }
 ```
 
+or implemented in Java
+
+```java
+@Override
+public IScope getScope(EObject context, EReference reference) {
+    // We want to define the Scope for the Element's superElement cross-reference
+    if (context instanceof Element
+            && reference == MyDslPackage.Literals.ELEMENT__SUPER_ELEMENT) {
```

Comment:
.equals?

Reply:
The EPackage contents object are singletons so identity equals is what I would consider idiomatic usage of the Literals interface.

Reply:
Alright, make sense